packages/flutter_tools/lib/src/base/build.dart

```diff
@@ -170,15 +195,13 @@ class AOTSnapshotter {
     genSnapshotArgs.add(mainPath);
 
     final SnapshotType snapshotType = SnapshotType(platform, buildMode);
-    final int genSnapshotExitCode =
-      await _timedStep('snapshot(CompileTime)', 'aot-snapshot',
```

Comment:
Is this timing info still captured some other way? Or is this reporting no longer valuable because we expect users to be taking a different path?

Reply:
We don't use a _timedStep  for the current AOT build.

All of this timing information is captured via PerformanceMeasurement classes. This isn't hooked up to analytics yet, but I could update this PR to do so. This has the benefit of allowing uniform handling of build performance

https://github.com/flutter/flutter/blob/master/packages/flutter_tools/lib/src/build_system/build_system.dart#L619

I could wire this up in this PR or do it separately

Reply:
Separately sgtm.

Reply:
Filled https://github.com/flutter/flutter/issues/52842 to track

Reply:
Removing this _timedStep removed the compilation times from the log output reported using the --report-timings flag in Golem benchmarking of Flutter.

The failure was not noticed, because the previous backup code path that was used before the --report-timings flag was introduced started running again.